commons-vfs2-jackrabbit2/src/main/java/org/apache/commons/vfs2/provider/webdav4/Webdav4FileObject.java

```diff
@@ -544,7 +544,8 @@ DavPropertySet getPropertyNames(final GenericURLFileName name) throws FileSystem
      * @return The encoded URL String.
      */
     private String hrefString(final GenericURLFileName name) {
-        final GenericURLFileName newFile = new GenericURLFileName("http", name.getHostName(), name.getPort(), name.getDefaultPort(),
+        final String scheme = getHttpScheme(name);
```

Comment:
`#getHttpScheme(name)` is unnecessary if you just use `getInternalURI().getScheme()` and move the line into try ~ catch block.

Reply:
Hi @woonsan ,
Thanks for suggestions! I have updated code as per your suggestion! please review again!